examples/examples/z_querier.rs

```diff
@@ -0,0 +1,155 @@
+//
+// Copyright (c) 2024 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+use std::time::Duration;
+
+use clap::Parser;
+use zenoh::{key_expr::KeyExpr, query::QueryTarget, Config};
+use zenoh_examples::CommonArgs;
+
+#[tokio::main]
+async fn main() {
+    // initiate logging
+    zenoh::init_log_from_env_or("error");
+    #[cfg(feature = "unstable")]
+    let (config, keyexpr, payload, target, timeout, add_matching_listener) = parse_args();
+    #[cfg(not(feature = "unstable"))]
+    let (config, keyexpr, payload, target, timeout, _) = parse_args();
+
+    println!("Opening session...");
+    let session = zenoh::open(config).await.unwrap();
+
+    println!("Declaring Querier on '{keyexpr}'...");
+    let querier = session
+        .declare_querier(keyexpr)
+        .target(target)
+        .timeout(timeout)
+        .await
+        .unwrap();
+
+    #[cfg(feature = "unstable")]
+    if add_matching_listener {
+        querier
+            .matching_listener()
+            .callback(|matching_status| {
```

Comment:
As this example uses streams for `get` it would make more sense to use streams for `matching_listener` as well with the help of a `tokio::select`.

Reply:
see response for publisher example above